src/main/java/com/kuzmins/model/BasicEntity.java

```diff
@@ -3,15 +3,22 @@
 
 
 import com.kuzmins.service.Printable;
+import jakarta.persistence.Column;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.MappedSuperclass;
 
 import java.util.UUID;
-
+@MappedSuperclass
 public abstract class BasicEntity implements Printable {
-
+    @Id
+    @Column(name="id")
+    @GeneratedValue(strategy = GenerationType.AUTO)
     protected UUID id;
 
     public BasicEntity(){
-        this.id = UUID.randomUUID();;
+        this.id = UUID.randomUUID();
     }
```

Comment:
when you use `@GeneratedValue(strategy = GenerationType.AUTO)` there is no need to set `id` in constructor, it will be generated on db level:
```java
protected BaseEntity() {}
```